rir/src/compiler/opt/cleanup.cpp

```diff
@@ -246,7 +246,7 @@ class TheCleanup {
 namespace rir {
 namespace pir {
 
-void Cleanup::apply(RirCompiler&, Closure* function) const {
+void Cleanup::apply(RirCompiler&, Closure* function, LogStream&) const {
```

Comment:
I don't see how this would simplify anything here. The analysis logger would then have to find the correct log stream (ie. the one that belongs to the function). Remember that this was one of the main issues with the stream logger manager thingy.